test/src/test_derive.rs

```diff
@@ -322,3 +327,120 @@ fn test_derive_nativeclass_without_constructor() -> bool {
 
     ok
 }
+
+#[derive(NativeClass)]
+#[inherit(Node)]
+struct CustomGetSet {
+    pub get_called: Cell<i32>,
+    pub set_called: Cell<i32>,
+    #[allow(dead_code)]
```

Comment:
It seems like we always need `#[allow(dead_code)]` whenever `Property` is involved (and obviously not accessed)?

I wonder if adding a
```rs
quote! {
    #[allow(dead_code)]
};
```
might do the job...

But this PR is already quite big, I'm also fine if we do this at some point in the future 🙂